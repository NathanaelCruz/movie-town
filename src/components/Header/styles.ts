import styled from 'styled-components'

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 9rem;
  background-color: ${(props) => props.theme.colors.black};
  padding: ${(props) =>
    props.theme.measures.small + ' ' + props.theme.measures.bigSmall};
`

export const HeaderContent = styled.div`
  display: grid;
  grid-template: 'logo busca busca busca busca';
  grid-gap: 1rem;
  width: 100%;
  height: 100%;
  max-width: 100rem;

  @media (min-width: 767px) {
    grid-template: 'logo busca busca . .';
  }
`

export const LinkLogo = styled.a`
  grid-area: logo;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 6.5rem;
`
export const Logo = styled.img`
  max-width: 100%;
`
