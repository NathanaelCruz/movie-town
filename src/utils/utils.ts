export const HostDataBase = 'http://localhost:3001'
