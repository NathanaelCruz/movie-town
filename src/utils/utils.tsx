import { ReactNode } from 'react'

import { ThemeProvider } from 'styled-components'
import theme from '../styles/theme'

export const HostDataBase = 'http://localhost:3001'

export function RenderWithTemplate(ComponentRender: ReactNode) {
  return <ThemeProvider theme={theme}>{ComponentRender}</ThemeProvider>
}
