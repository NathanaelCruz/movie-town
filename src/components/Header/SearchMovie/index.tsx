import * as S from './styles'
import { SavedSearch } from '@styled-icons/material'
import { KeyboardEvent, useContext } from 'react'
import { ListMoviesContext } from '../../../contexts/ListMoviesContext'

const SearchMovie: React.FC = () => {
  const { listMoviesSearch, searchForWordKey } = useContext(ListMoviesContext)

  function searchWordKey(e: KeyboardEvent) {
    const valueCurrency = e.target as HTMLInputElement
    listMoviesSearch(valueCurrency.value)
  }

  return (
    <S.SearchContent>
      <S.SearchInput
        type="search"
        placeholder="Qual seu filme favorito?"
        onKeyUp={(event) => searchWordKey(event)}
      ></S.SearchInput>
      <S.SearchButton onClick={() => searchForWordKey()}>
        <SavedSearch size={24} color="#ffffff"></SavedSearch>
      </S.SearchButton>
    </S.SearchContent>
  )
}

export default SearchMovie
