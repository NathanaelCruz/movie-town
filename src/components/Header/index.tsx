import * as S from './styles'
import Link from 'next/link'
import SearchMovie from './SearchMovie'

const Header: React.FC = () => {
  return (
    <S.Header>
      <S.HeaderContent role="menu">
        <Link href="/">
          <S.LinkLogo>
            <S.Logo
              src="/img/icon-100.png"
              alt="Ticket de ingresso para o cinema com uma estrela"
            />
          </S.LinkLogo>
        </Link>
        <SearchMovie />
      </S.HeaderContent>
    </S.Header>
  )
}

export default Header
