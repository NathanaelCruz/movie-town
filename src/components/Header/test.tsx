import { render, screen } from '@testing-library/react'
import { RenderWithTemplate } from '../../utils/utils'
import Header from './index'

describe('Header', () => {
  it('Deverá renderizar o Header', () => {
    const { container } = render(RenderWithTemplate(<Header />))

    expect(container).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('Deverá ser renderizado um loogo', () => {
    render(RenderWithTemplate(<Header />))

    expect(
      screen.getByRole('img', {
        name: /Ticket de ingresso para o cinema com uma estrela/i
      })
    ).toBeInTheDocument()
  })
})
