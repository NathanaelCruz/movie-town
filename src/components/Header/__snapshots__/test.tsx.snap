// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Dever√° renderizar o Header 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 9rem;
  background-color: #2f3542;
  padding: 1.3rem 0.9rem;
}

.c1 {
  display: grid;
  grid-template: 'logo busca busca busca busca';
  grid-gap: 1rem;
  width: 100%;
  height: 100%;
  max-width: 100rem;
}

.c2 {
  grid-area: logo;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 6.5rem;
}

.c3 {
  max-width: 100%;
}

.c4 {
  grid-area: busca;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-height: 100%;
  position: relative;
}

.c5 {
  width: 95%;
  height: 3rem;
  padding: 1rem 4rem 1rem 1rem;
  border-radius: 15px 0 0 15px;
  border: none;
  outline: none;
}

.c5:active,
.c5:focus {
  box-shadow: 0 0 5px 2px #E74C3C;
}

.c6 {
  background-color: #2f3542;
  position: absolute;
  height: 4rem;
  width: 4rem;
  z-index: 10;
  right: 0;
  top: calc(50% - 2rem);
  border-radius: 50%;
  border: none;
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
}

.c6:hover {
  cursor: pointer;
  background-color: #E74C3C;
}

.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (min-width:767px) {
  .c1 {
    grid-template: 'logo busca busca . .';
  }
}

<header
  class="c0"
>
  <div
    class="c1"
    role="menu"
  >
    <a
      class="c2"
    >
      <img
        alt="Ticket de ingresso para o cinema com uma estrela"
        class="c3"
        src="/img/icon-100.png"
      />
    </a>
    <div
      class="c4"
    >
      <input
        class="c5"
        placeholder="Qual seu filme favorito?"
        type="search"
      />
      <button
        class="c6"
      >
        <svg
          aria-hidden="true"
          class="c7"
          color="#ffffff"
          fill="currentColor"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14zm-2.17-1.5l2.14-1.53 2.14 1.53-.83-2.46 2.15-1.5h-2.62L9.47 6l-.84 2.54H6l2.14 1.49z"
          />
        </svg>
      </button>
    </div>
  </div>
</header>
`;
